agentops/decorators.py

```diff
@@ -67,10 +67,6 @@ async def async_wrapper(*args, session: Optional[Session] = None, **kwargs):
                 try:
                     returns = await func(*args, **kwargs)
 
-                    # If the function returns multiple values, record them all in the same event
-                    if isinstance(returns, tuple):
-                        returns = list(returns)
-
```

Comment:
Instead of removing these functions, we can instead do what we intended at fist-- just return a tuple

```
if istinatnce(returns, tuple):
    returns = tuple(returns)
```

Removing these lines of code altogether will have unintended consequences.

Reply:
May I ask what the suggested change does? It seems (nearly) vacuous to me, since you're casting a tuple to the same tuple.

Also, I'm curious about the unintended consequences if the lines were removed--when I removed them my code seemed to work well.

Unfortunately your suggested change would still break the code I'm using, since the code checks the output of the function is an instance of a *specific subclass* of `tuple` (more specifically, a custom `namedtuple`). Maybe one could argue that this is bad practice, but I believe it's more important that a decorator that does logging doesn't change the output of the function.

Reply:
> but AFAIK the only reason the tuple was being cast to a list was so that it can be dumped into a JSON, but json.dumps actually handles that automatically.

Indeed that should be handled by a `JSONEncoder` which is where these sorts of casts belong